module-system/module-implementations/sov-bank/tests/gas_test.rs

```diff
@@ -0,0 +1,223 @@
+use helpers::*;
+use sov_bank::{
+    get_genesis_token_address, Bank, BankConfig, BankGasConfig, CallMessage, TokenConfig,
+};
+use sov_modules_api::macros::config_constant;
+use sov_modules_api::utils::generate_address;
+use sov_modules_api::{Context, GasUnit, Module, WorkingSet};
+use sov_prover_storage_manager::new_orphan_storage;
+use tempfile::TempDir;
+
+mod helpers;
+
+const CREATE_TOKEN_NATIVE_PRICE: u64 = 2;
+const CREATE_TOKEN_ZK_PRICE: u64 = 3;
+
+#[test]
+fn zeroed_price_wont_deduct_working_set() {
+    let sender_balance = 100;
+    let remaining_funds = BankGasTestCase::init(sender_balance).execute().unwrap();
+
+    assert_eq!(
+        remaining_funds, sender_balance,
+        "the balance should be unchanged with zeroed price"
+    );
+}
+
+#[test]
+fn normal_price_will_deduct_working_set() {
+    let sender_balance = 100;
+
+    let native_price = 2;
+    let zk_price = 3;
+    let remaining_funds = BankGasTestCase::init(sender_balance)
+        .with_native_price(native_price)
+        .with_zk_price(zk_price)
+        .override_gas_config()
+        .execute()
+        .unwrap();
+
+    // compute the expected gas cost, based on the test constants
+    let gas_cost = native_price * CREATE_TOKEN_NATIVE_PRICE + zk_price * CREATE_TOKEN_ZK_PRICE;
```

Comment:
Same as below `CREATE_TOKEN_X_PRICE` should be rnamed to soemthing like `CREATE_TOKEN_X_COST `

Reply:
Fixed!